src/main/java/org/folio/service/pieces/flows/update/PieceUpdateFlowManager.java

```diff
@@ -61,22 +61,29 @@ public Future<Void> updatePiece(Piece pieceToUpdate, boolean createItem, boolean
       })
       .compose(title -> protectionService.isOperationRestricted(holder.getTitle().getAcqUnitIds(), UPDATE, requestContext))
       .compose(v -> pieceUpdateFlowInventoryManager.processInventory(holder, requestContext))
-      .compose(vVoid -> updatePoLine(holder, requestContext))
-      .map(afterUpdate -> {
-        JsonObject messageToEventBus = new JsonObject();
-        messageToEventBus.put("poLineIdUpdate", holder.getPieceToUpdate().getPoLineId());
+      .compose(v -> updatePoLine(holder, requestContext))
+      .map(v -> {
         Piece.ReceivingStatus receivingStatusStorage = holder.getPieceFromStorage().getReceivingStatus();
         Piece.ReceivingStatus receivingStatusUpdate = holder.getPieceToUpdate().getReceivingStatus();
         logger.debug("receivingStatusStorage -- {}", receivingStatusStorage);
         logger.debug("receivingStatusUpdate -- {}", receivingStatusUpdate);
         if (receivingStatusStorage.compareTo(receivingStatusUpdate) != 0) {
           holder.getPieceToUpdate().setStatusUpdatedDate(new Date());
+          return true;
+        }
+        return false;
+      })
+      .compose(verifyReceiptStatus -> pieceStorageService.updatePiece(holder.getPieceToUpdate(), requestContext)
+        .map(verifyReceiptStatus))
+      .map(verifyReceiptStatus -> {
+        if (verifyReceiptStatus) {
```

Comment:
this line is confusing to reader. it would be better to use Boolean.TRUE.equals(verify...)

Reply:
also this is causing sonarqube issue

Reply:
I noticed the warning, but I think this is an error of the tool: given the code just before, values for `verifyReceiptStatus` can only be `TRUE` or `FALSE`, not `null`. And I think "`if (verifyReceiptStatus)`" is much more readable than "`if (Boolean.TRUE.equals(verifyReceiptStatus))`". But since @azizbekxm finds it confusing, I will switch to using `Boolean.TRUE.equals()`.

Reply:
I agree your point of view, if (isChecked..) is more readable but verifyReceiptStatus is not implying boolean type is more like action so that is why I give recommendation to use Boolean.TRUE.equals().